src/build.d

```diff
@@ -1867,6 +1899,80 @@ auto run(T)(T args, string workDir = runDir)
     return res.output;
 }
 
+/**
+Wrapper around execute that logs the execution, prints output from stderr,
+and throws an exception for a non-zero exit code.
+
+Params:
+    args = the command and command arguments to execute
+    workDir = the commands working directory
+*/
+void verboseRun(T)(string section, T args, string workDir = runDir)
+{
+    import std.ascii : newline;
+
+    args = args.filter!(a => !a.empty).array;
+    log("Run: %s", args.join(" "));
+    auto p = pipeProcess(args, Redirect.stdout | Redirect.stderr, null,
+                           Config.none, workDir);
+
+    auto appStdErr = appender!string;
+    auto appStdOut = appender!string;
+    enum size_t chunkSize = 4096;
+
+    // consume a stream into an appender without limit
+    static void consumeStream(A, R)(ref A app, ref R r)
+    {
+        if (!r.empty)
+        {
+            auto chunk = r.front;
+            r.popFront;
+            app.put(chunk);
+        }
+    }
+
+    foreach (_; 0..2) // make sure both streams are exhausted
+    {
+        auto stdoutChunks = p.stdout.byChunk(chunkSize);
+        auto stderrChunks = p.stderr.byChunk(chunkSize);
+        while (!stdoutChunks.empty || !stderrChunks.empty)
+        {
+            consumeStream(appStdOut, stdoutChunks);
+            consumeStream(appStdErr, stderrChunks);
+        }
+    }
```

Comment:
This seems duplicated, maybe extend `consumeStream` to read an entire pipe s.t. you can simply do below:
```
auto dataStdout = consmeStream(p.stdout);
auto dataStderr = consmeStream(p.stderr);
```

That would also make it easier to read those pipes lazily depending on the control flow below.

Reply:
If we wait until stdout gets closed and then only start reading from stderr, we risk that the program wrote too much than the maximum size of the pipe.